packages/components/data-table/src/data-row.js

```diff
@@ -53,6 +47,13 @@ const DataRow = (props) => {
             props.columns.length,
             columnIndex
           )}
+          onCellClick={
+            props.onRowClick && !column.shouldIgnoreRowClick
+              ? () =>
+                  props.onRowClick &&
+                  props.onRowClick(props.row, props.rowIndex, column.key)
```

Comment:
```suggestion
                  props.onRowClick?.(props.row, props.rowIndex, column.key)
```

Should work with optional chaining 🤔 . Syntax looks really odd but I think it was it.

Reply:
Ah nice! I'm committing this and let the tests be the judge :)